kontrol-service/engine/docker.go

```diff
@@ -259,21 +327,60 @@ func newStatefulPluginsAndExternalServicesFromServiceConfig(serviceConfig apityp
 	return serviceStatefulPlugins, externalServices, externalServiceDependencies, nil
 }
 
-func newClusterTopologyServiceFromServiceConfig(serviceConfig apitypes.ServiceConfig, version string) resolved.Service {
+func newClusterTopologyServiceFromServiceConfig(
+	serviceConfig apitypes.ServiceConfig,
+	deploymentConfig *apitypes.DeploymentConfig,
+	statefulSetConfig *apitypes.StatefulSetConfig,
+	version string,
+) (resolved.Service, error) {
 	service := serviceConfig.Service
-	deployment := serviceConfig.Deployment
+	serviceName := service.GetObjectMeta().GetName()
+
+	if deploymentConfig == nil && statefulSetConfig == nil {
```

Comment:
I think you can remove this validation because there is another one below `clusterTopologyService.WorkloadSpec == nil`

Reply:
actually the one below should be removed